app/web3/transactionConfirmation.ts

```diff
@@ -0,0 +1,73 @@
+/* tslint:disable:no-console */
+
+import config from "../config";
+import web3Provider from "./web3Provider";
+
+export const transactionConfirmation = async (
+  transactionHash: string,
+  transactionMinedCallback: any,
+  newBlockCallback: any
+) => {
+  console.log("waiting for transaction: " + transactionHash);
+  return new Promise((resolve, reject) => {
+    let prevBlockNo = -1;
+    let startingBlock = -1;
+    const requiredConfirmations = config.transactionSigning.numberOfConfirmations;
+    const maxNumberOfBlocksToWait = config.transactionSigning.maxNumberBlocksToWait;
+    const poll = async () => {
+      let currentBlockNo;
+      try {
+        currentBlockNo = await promisify(web3Provider.eth.getBlockNumber, []);
+      } catch (e) {
+        console.log("error in web3.eth.getBlockNumber");
+        console.log(e);
+        return reject(e);
+      }
+
+      if (startingBlock === -1) {
+        startingBlock = currentBlockNo;
+      }
+
+      if (currentBlockNo - startingBlock >= maxNumberOfBlocksToWait) {
+        return reject(
+          `Your transaction has not been mined in last ${maxNumberOfBlocksToWait} blocks`
+        );
+      }
+
+      console.log(`got block number ${currentBlockNo} prev block number ${prevBlockNo}`);
+      if (currentBlockNo !== prevBlockNo) {
+        prevBlockNo = currentBlockNo;
+        newBlockCallback(currentBlockNo);
+
+        try {
+          const transaction = await promisify(web3Provider.eth.getTransaction, [transactionHash]);
```

Comment:
please promisify all functions at the root of this file first and then just use them.